pennylane/ops/op_math/exp.py

```diff
@@ -310,7 +312,7 @@ def _recursive_decomposition(self, base: Operator, coeff: complex):
                 # Some generators are not wire-ordered (e.g. OrbitalRotation)
                 mapped_wires_g = qml.map_wires(g, dict(zip(g.wires, base.wires)))
 
-                if qml.equal(base, mapped_wires_g) and math.real(coeff) == 0:
+                if qml.equal(mapped_wires_g, base) and math.real(coeff) == 0:
```

Comment:
As mentioned in the meeting, this somehow fixes things.

Reply:
🤯

Reply:
One potential reasons is that equal checks the isinsatnce(one, type(other)). So while a LinearCombination is a Sum, the other way around would give False. Not sure that is the thing here or if it is something else, there are a lot of things in qml.equal 😅

Reply:
> One potential reasons is that equal checks the isinsatnce(one, type(other)). So while a LinearCombination is a Sum, the other way around would give False.

That is exactly the reason.

Reply:
I understand the reason, that doesn't answer my question. To me this looks like your are working around a bug - the type of the operator returned my `qml.map_wires` shouldn't be different than the input operator as far as I know. So the question is:

> What class of operator is changing class when we call qml.map_wires on it?

Reply:
Unless `g` is a different class than `base` from the get-to. I just want to be sure I understand what's happening here.

Reply:
This change was due to a previous decision (reverted now as suggested) to keep `LinearCombination` as `LinearCombination` instead of always changing it to `Sum` using `qml.dot`, whereas the `qml.generator` from before returns a `Sum`, which caused the `equals` check to fail.

Reply:
Awesome, thanks. I was having trouble seeing why `g` and `base` would differ, so I was concerned the difference was arising in `wire_mapping` somehow 😅  Glad that's not the case!